### PR TITLE
fix: check all matches when looking for rule IDs

### PR DESCRIPTION
**Pull request by theseion**

### CODE REVIEW

**Comment by fzipi** [comment]
Should we add some tests here?

**Comment by theseion** [comment]
Done.

File: waflog/read.go [hunk 1]
```diff
@@ -28,16 +28,21 @@ func (ll *FTWLogLines) TriggeredRules() []uint {
...
+		match := regex.FindAllSubmatch(line, -1)
 		if match != nil {
 			log.Trace().Msgf("ftw/waflog: Found %s at %s", regex.String(), line)
-			submatch := string(match[1])
-			ruleId, err := strconv.ParseUint(submatch, 10, 0)
-			if err != nil {
-				log.Error().Msgf("Failed to parse uint from %s", submatch)
-				continue
+			for _, nextMatch := range match {
+				submatch := string(nextMatch[1])
```
Comment (huberts90) [inline waflog/read.go:35]: Could you please check if the `len(nextMatch) > 0` firstly?

File: waflog/read.go [hunk 2]
```diff
@@ -28,16 +28,21 @@ func (ll *FTWLogLines) TriggeredRules() []uint {
...
+				if len(submatch) == 0 {
+					submatch = string(nextMatch[2])
```
Comment (huberts90) [inline waflog/read.go:37]: Could you please check if the `len(nextMatch) > 1` firstly?

File: waflog/read.go (hunk 1 above)
Comment (theseion) [reply waflog/read.go:35]: Unnecessary. When the expression matches successfully, `nextMatch` cannot be empty. The submatches are not optional.

File: waflog/read.go (hunk 2 above)
Comment (theseion) [reply waflog/read.go:37]: Same as above.

File: waflog/read.go (hunk 1 above)
Comment (theseion) [reply waflog/read.go]: I've replaced the explicit accesses with a loop. Thanks for the hint.

**Pull request merged**